src/transformers/models/mask_rcnn/modeling_maskrcnn.py

```diff
@@ -0,0 +1,3818 @@
+        elif reduction != "none":
+            raise ValueError('avg_factor can not be used with reduction="sum"')
+    return loss
+
+
+def accuracy(pred, target, topk=1, thresh=None):
+    """Calculate accuracy according to the prediction and target.
+
+    Args:
+        pred (`torch.Tensor`):
+            The model prediction, shape (N, num_class).
+        target (`torch.Tensor`):
+            The target of each prediction, shape (N, ).
+        topk (`int` or `Tuple[int]`, *optional*, defaults to 1):
+            If the predictions in `topk` matches the target, the predictions will be regarded as correct ones.
+        thresh (`float`, *optional*):
+            If not None, predictions with scores under this threshold are considered incorrect.
+
+    Returns:
+        `float` or `Tuple[float]`: If the input `topk` is a single integer,
+            the function will return a single float as accuracy. If `topk` is a tuple containing multiple integers, the
+            function will return a tuple containing accuracies of each `topk` number.
+    """
+    if not isinstance(topk, (int, tuple)):
+        raise ValueError(f"topk should be int or tuple of int, but got {type(topk)}")
+    if isinstance(topk, int):
+        topk = (topk,)
+        return_single = True
+    else:
+        return_single = False
+
+    maxk = max(topk)
+    if pred.size(0) == 0:
+        accu = [pred.new_tensor(0.0) for i in range(len(topk))]
+        return accu[0] if return_single else accu
+    if not (pred.ndim == 2 and target.ndim == 1):
+        raise ValueError(
+            f"pred and target should be of shape (N, C) and (N, ), but got {pred.shape} and {target.shape}"
+        )
+    if pred.size(0) != target.size(0):
+        raise ValueError(f"pred and target should be of same size, but got {pred.size(0)} and {target.size(0)}")
+    if maxk > pred.size(1):
+        raise ValueError(f"maxk {maxk} exceeds pred dimension {pred.size(1)}")
+    pred_value, pred_label = pred.topk(maxk, dim=1)
+    pred_label = pred_label.t()  # transpose to shape (maxk, N)
+    correct = pred_label.eq(target.view(1, -1).expand_as(pred_label))
+    if thresh is not None:
+        # Only prediction values larger than thresh are counted as correct
+        correct = correct & (pred_value > thresh).t()
+    res = []
+    for k in topk:
+        correct_k = correct[:k].reshape(-1).float().sum(0, keepdim=True)
+        res.append(correct_k.mul_(100.0 / pred.size(0)))
+    return res[0] if return_single else res
+
+
+def cross_entropy(
+    pred,
+    label,
+    weight=None,
+    reduction="mean",
+    avg_factor=None,
+    class_weight=None,
+    ignore_index=-100,
+    avg_non_ignore=False,
+):
+    """Calculate the CrossEntropy loss.
+
+    Args:
+        pred (`torch.Tensor`):
+            The prediction with shape (N, C), C is the number of classes.
+        label (`torch.Tensor`):
+            The learning label of the prediction.
+        weight (`torch.Tensor`, *optional*):
```

Comment:
```suggestion
        sample_weight (`torch.Tensor`, *optional*):
```

Reply:
Would use the same term as PyTorch's one: https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html